libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -832,7 +832,7 @@ string ABIFunctions::abiEncodingFunctionStruct(
 			}
 			memberTempl("encodingOffset", toCompactHexWithPrefix(encodingOffset));
 			encodingOffset += dynamicMember ? 0x20 : memberTypeTo->calldataEncodedSize();
-			memberTempl("abiEncode", abiEncodingFunction(*memberTypeFrom, *memberTypeTo, _encodeAsLibraryTypes, false));
+			memberTempl("abiEncode", abiEncodingFunction(*memberTypeFrom, *memberTypeTo, _encodeAsLibraryTypes, true));
```

Comment:
The `true` here seems to refer to the `_compacted` parameter which only seems to be used for function type encoding.  Which then seems to mean whether the function type is a single stack item or two items (addr, selector), where true means single.

Is that correct? Takes quite a lot to trace it and hence understanding what is going on.

Reply:
Yes, that's true. Unfortunately, we have to pass this parameter through all calls. Do you have a better idea?

Reply:
Why do we have two versions of the stack for this type?

Reply:
If the type is stored on the stack, it always uses two slots, but if it is stored in memory, those are combined. It makes sense to not split and recombine it in the ABI encoding if it is just copied somwhere, for example if you want to encode a memory array of function type.

Reply:
So perhaps a better name instead of `compact` is `fromMemory`/`fromStack` ?

Reply:
Yes, I think I did something like that in the decoder.

Reply:
Can you rename it? If not int his PR in another one before we forget.

Reply:
Renamed.